tunnel/winipcfg/types.go

```diff
@@ -944,13 +948,28 @@ func (row *MibIPforwardRow2) Delete() error {
 }
 
 // get method returns all table rows as a Go slice.
-func (tab *mibIPforwardTable2) get() []MibIPforwardRow2 {
-	const maxCount = maxIndexCount8 / unsafe.Sizeof(MibIPforwardRow2{})
-	return (*[maxCount]MibIPforwardRow2)(unsafe.Pointer(&tab.table[0]))[:tab.numEntries]
+func (tab *mibIPforwardTable2) get() (s []MibIPforwardRow2) {
+	h := (*unsafeSlice)(unsafe.Pointer(&s))
+	h.Data = unsafe.Pointer(&tab.table[0])
+	h.setLen(int(tab.numEntries))
+	return
 }
 
 // free method frees the buffer allocated by the functions that return tables of network interfaces, addresses, and routes.
 // https://docs.microsoft.com/en-us/windows/desktop/api/netioapi/nf-netioapi-freemibtable
 func (tab *mibIPforwardTable2) free() {
 	freeMibTable(unsafe.Pointer(tab))
 }
+
+type unsafeSlice struct {
```

Comment:
There's reflect.SliceHeader. I noticed that sys doesn't use this either, though. Is this because it's not guaranteed to be correct, whereas reflect must stay stable? Or because the linker stops optimizing things when it sees reflect has been used? Or because that data pointer is of an annoying type there? Or some other reason?

Reply:
Yeah, the `Data uintptr` predates Go having a precise garbage collector aware of all the types of objects in memory, and dates back to a time when we tried to minimize the use of `unsafe` in part due to App Engine limitations. But mostly an early oversight we couldn't fix.

But we didn't want to add `reflect.SliceHeader2`. Some wanted to put it in unsafe itself, or runtime, etc, but nobody could agree, so it lives in an internal package in x/sys and copied around (like here) when needed with the correct types.

Reply:
Well, let's see what happens this time around: https://go-review.googlesource.com/c/go/+/263800

Reply:
See why I added this now? :)

Reply:
Ooof. Yup!